src/Config/ConfigurationStore.php

```diff
@@ -2,170 +2,50 @@
+class ConfigurationStore
 {
-    private CacheInterface $flagCache;
-    private CacheInterface $banditCache;
-    private CacheInterface $metadataCache;
+    private const CONFIG_KEY = "EPPO_configuration_v1";
+    private ?Configuration $configuration = null;
 
-    // Key for storing bandit variations in the metadata cache.
-    private const BANDIT_VARIATION_KEY = 'banditVariations';
-
-    /**
-     * @param CacheInterface $cache
-     */
-    public function __construct(CacheInterface $cache)
+    public function __construct(private readonly CacheInterface $cache)
     {
-        $this->flagCache = new NamespaceCache(CacheType::FLAG, $cache);
-        $this->banditCache = new NamespaceCache(CacheType::BANDIT, $cache);
-        $this->metadataCache = new NamespaceCache(CacheType::META, $cache);
     }
 
-    /**
-     * @param string $key
-     * @return Flag|null
-     */
-    public function getFlag(string $key): ?Flag
+    public function getConfiguration(): Configuration
     {
-        try {
-            $result = $this->flagCache->get($key);
-            if ($result == null) {
-                return null;
-            }
-
-            $inflated = $result;
-            return $inflated === false ? null : $inflated;
-        } catch (\Psr\SimpleCache\InvalidArgumentException $e) {
-            // Simple cache throws exceptions when a keystring is not a legal value (characters {}()/@: are illegal)
-            syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key ${key}: " . $e->getMessage());
-            return null;
-        }
-    }
+        if ($this->configuration === null) {
+            try {
+                $cachedConfig = $this->cache->get(self::CONFIG_KEY);
+                if (!$cachedConfig) {
+                    return Configuration::emptyConfig(); // Empty config
+                }
 
-    /**
-     * @param array $flags
-     * @param IBanditReferenceIndexer|null $banditVariations
-     * @throws EppoClientException
-     */
-    public function setUnifiedFlagConfiguration(array $flags, ?IBanditReferenceIndexer $banditVariations = null): void
-    {
-        try {
-            // Clear stored config before setting data.
-            $this->flagCache->clear();
+                $json = json_decode($cachedConfig, true);
+                if ($json === null) {
+                    return Configuration::emptyConfig();
+                }
 
-            $this->setFlags($flags);
-            if ($banditVariations == null) {
-                $this->metadataCache->delete(self::BANDIT_VARIATION_KEY);
-            } else {
-                $this->metadataCache->set(self::BANDIT_VARIATION_KEY, $banditVariations);
+                $configurationWire = ConfigurationWire::fromJson($json);
+                $this->configuration = Configuration::fromConfigurationWire($configurationWire);
+            } catch (InvalidArgumentException $e) {
+                // Safe to ignore as the const `CONFIG_KEY` contains no invalid characters
```

Comment:
FOR NOW!

But I do wonder if we should still log _something_

Reply:
done